test/modules/shadowing/issue-19167-7-x.chpl

```diff
@@ -0,0 +1,21 @@
+module A {
+  proc f() { writeln("A.f"); }
+  var x = "A";
+}
+module B {
+  proc f() { writeln("B.f"); }
+  var x = "B";
+}
+
+module UseA_ImportB {
+  public use A;
+  import B;
+}
+
+module Program {
+  use UseA_ImportB;
+
+  proc main() {
+    writeln(x);
+  }
+}
```

Comment:
All these 19167 behaviors are looking good to me.  Do we have a case in which the _same_ function f() or variable `x` is reached through two distinct public use chains that take different paths to get back to the same module, to make sure that we're not treating the two paths as though they were creating two different functions/symbols?  I feel pretty confident that this would work (it seems likely that it probably happens all the time through standard and internal modules?), but it's a case that I wondered about while reading through these variations, so might be good to lock in.  I'm imagining something like:

```chapel
module A {
  proc f() { writeln("A.f"); }
  var x = "A";
}
module B {
  public use A;
}
module C {
  public use A;
}
module Program {
  use B, C;
  f();
  writeln(x);
}
```

Reply:
Yeah, there's some pre-existing visibility tests that exercise this and didn't look broken by the changes Michael made in this PR

Reply:
Yes, based on @lydia-duncan's point I don't plan to add a test since it should be covered, but I did spot-check that particular program and it runs as I would expect.